pkg/reconciler/route/resources/virtual_service.go

```diff
@@ -145,50 +101,68 @@ func buildHTTPRoute(hostDomain, namespace, urlPath string, appNames []string) ([
 		})
 	}
 
-	var httpRoutes []networking.HTTPRoute
-
-	for _, appName := range appNames {
-		httpRoutes = append(httpRoutes, networking.HTTPRoute{
-			Match: pathMatchers,
-			Route: buildRouteDestination(),
-			Rewrite: &networking.HTTPRewrite{
-				Authority: network.GetServiceHostname(appName, namespace),
-			},
-			Headers: &networking.Headers{
-				Request: &networking.HeaderOperations{
-					Add: map[string]string{
-						// Set forwarding headers so the app gets the real hostname it's serving
-						// at rather than the internal one:
-						// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded1
-						"X-Forwarded-Host": hostDomain,
-						"Forwarded":        fmt.Sprintf("host=%s", hostDomain),
+	if len(routes) != 0 {
+		return []networking.HTTPRoute{
+			{
+				Match: pathMatchers,
+				Route: buildRouteDestinations(routes),
+				Headers: &networking.Headers{
+					Request: &networking.HeaderOperations{
+						Add: map[string]string{
+							// Set forwarding headers so the app gets the real hostname it's serving
+							// at rather than the internal one:
+							// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded1
+							"X-Forwarded-Host": hostDomain,
+							"Forwarded":        fmt.Sprintf("host=%s", hostDomain),
+						},
 					},
 				},
 			},
-		})
+		}, nil
 	}
 
-	// If there aren't any services bound to the route, we just want to
-	// serve a 503.
-	if len(httpRoutes) == 0 {
-		return []networking.HTTPRoute{
-			{
-				Match: pathMatchers,
-				Fault: &networking.HTTPFaultInjection{
-					Abort: &networking.InjectAbort{
-						Percent:    100,
-						HTTPStatus: http.StatusServiceUnavailable,
+	// If there are no apps mapped to this hostdomain+path combo, return HTTP route with fault
+	return []networking.HTTPRoute{
+		{
+			Match: pathMatchers,
+			Fault: &networking.HTTPFaultInjection{
+				Abort: &networking.InjectAbort{
+					Percent:    100,
+					HTTPStatus: http.StatusServiceUnavailable,
+				},
+			},
+			Route: buildDefaultRouteDestination(),
+		},
+	}, nil
+}
+
+func buildRouteDestinations(routes []*v1alpha1.Route) []networking.HTTPRouteDestination {
+
+	routeWeights := getRouteWeights(len(routes))
+	routeDestinations := []networking.HTTPRouteDestination{}
+	for i, route := range routes {
+		namespace := route.GetNamespace()
+		appName := route.Spec.AppName
+		routeDestination := networking.HTTPRouteDestination{
+			Destination: networking.Destination{
+				Host: GatewayHost,
+			},
+			Headers: &networking.Headers{
+				Request: &networking.HeaderOperations{
+					Set: map[string]string{
+						"Host": network.GetServiceHostname(appName, namespace),
```

Comment:
Would you mind adding tests to ensure this new field is covered? It's strange that none were broken from this change

Reply:
done